icp:
  # Registration method
  registration_method: gicp
  # Stop ICP if the transformation from the last iteration was this small.
  tf_epsilon: 0.001

  # During ICP, two points won't be considered a correspondence if they are at
  # least this far from one another.
  corr_dist: 1.0

  # Iterate ICP this many times.
  iterations: 20 # Default from original branch is 20

  # Maximum acceptable incremental rotation and translation.
  transform_thresholding: true
  max_translation: 1.0
  max_rotation: 1.0

  # Number of threads GICP can use (3 threads ~ 3x speedup)
  # num_threads: 2 - defined in launch files to adapt to robot

  # Enable GICP timing output
  enable_timing_output: false

# in general if there is no particular reason it should always be false,
# since it recomputes the covariances from scratch but we calculate them from normals
  recompute_covariances: false
